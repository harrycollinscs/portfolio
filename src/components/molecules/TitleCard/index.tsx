import React from 'react'
import Card from "../../atoms/Card"
import styled from 'styled-components'

const Content = styled.div`
display: flex;
flex-direction: column;
justify-content: center;
height: 100%;

h1 {
  color: ${(props) => props.color};
  font-size: 300%;

  @media screen and (max-width: 450px) {
    font-size: 250%;
  }
  @media screen and (max-width: 380px) {
    font-size: 200%;
  }
  @media screen and (max-width: 320px) {
    font-size: 150%;
  }
}
`

const TitleCard = ({ title, bgColor, color }: { title: string, bgColor: string, color: string }) => (
  <Card bgColor={bgColor} long>
    <Content color={color}>
      <h1>{title}</h1>
    </Content>
  </Card>
)

export default TitleCard