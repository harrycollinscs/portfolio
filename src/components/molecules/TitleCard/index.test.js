import React from 'react'
import renderer from 'react-test-renderer'
import TitleCard from './index'

it('renders correctly', () => {
  const tree = renderer.create(<TitleCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
