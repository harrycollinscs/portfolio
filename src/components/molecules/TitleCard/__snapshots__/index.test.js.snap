// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bcXHqe jezpzi"
>
  <div
    className="sc-gswNZR gPXlaZ"
  >
    <h1 />
  </div>
</div>
`;
